### PR TITLE
[GH-7494] Implement the system_admin and inactive filters for users under system console

### PR DESCRIPTION
**Pull request by pradeepmurugesan**
#### Summary
Changes to filter the user in system console. Implemented the inactive filter and system admin filter

#### Ticket Link
https://github.com/mattermost/mattermost-server/issues/7494

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)
- [x] Has server changes (https://github.com/mattermost/mattermost-server/pull/9976)
- [x] Has redux changes (https://github.com/mattermost/mattermost-redux/pull/747)
- [x] Has UI changes
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/mattermost-webapp/blob/master/i18n/en.json)) updates

screen shot 2018-12-24 at 19 54 17

screen shot 2018-12-24 at 19 54 29

### CODE REVIEW

**Comment by jasonblais** [comment]
Awaiting for server and redux PRs to merged before testing webapp changes.

**Comment by wiersgallak** [comment]
This is great @pradeepmurugesan!  Thank you so much.  I agree, having the "Filter:" label changed to "Role:" makes is more obvious about what users can expect from this feature.

**Comment by pradeepmurugesan** [comment]
Hello @jasonblais and @wiersgallak thanks for reviewing. 

I named it as `filter` based on the conversation we had [here](https://pre-release.mattermost.com/core/pl/31u5xnet5ir8udrorkznppngdh)

Kindly let me know if we need to still have it as `role`. I can change it. 

Thanks,
Pradeep

**Comment by jasonblais** [comment]
@pradeepmurugesan @wiersgallak I propose we use `Role`, or some other more descriptive term than `Filter`. Having a `Team` and `Filter` options isn't obvious.

We have a `Manage Roles` menu option on the Users page, so I don't think it's going to lead to too much confusion with the current options - the `inactive` option doesn't really fit, but it's close enough.

I feel strongly that `Filter` will be confusing to the System Admin.

image

**Comment by wiersgallak** [comment]
@pradeepmurugesan sorry for the back and forth.  Jason and I discussed and came up with ```User Status:``` as a replacement for ```Filter:```.  We both feel this is a better choice than role based on the prior conversation with the development team and will accommodate the additional filters want to include in the future.  
//cc: @jasonblais

**Comment by pradeepmurugesan** [comment]
@jasonblais @wiersgallak Fixed the same 👍

**Comment by wiersgallak** [comment]
Looks great -thank you! I will add a couple of developers for a review of the code.

File: components/admin_console/system_users/system_users.jsx [hunk 1]
```diff
@@ -132,19 +136,25 @@ export default class SystemUsers extends React.Component {
...
-        this.loadDataForTeam(teamId);
-        this.props.actions.setSystemUsersSearch(this.props.searchTerm, teamId);
+        this.loadDataForTeam(teamId, this.props.filter);
+        this.props.actions.setSystemUsersSearch(this.props.searchTerm, teamId, this.props.filter);
+    }
+
+    handleFilterChange(e) {
+        const filter = e.target.value;
+        this.props.actions.setSystemUsersSearch(this.props.searchTerm, this.props.teamId, filter);
+        this.loadDataForTeam(this.props.teamId, filter);
```
Comment (lieut-data) [inline components/admin_console/system_users/system_users.jsx]: nit: these two lines are swapped between the function above. Maybe swap back for consistency?

File: components/admin_console/system_users/system_users.jsx [hunk 2]
```diff
@@ -203,6 +212,15 @@ export default class SystemUsers extends React.Component {
...
 
+    getFilterOptions(filter) {
+        const options = {};
+        if (filter === SearchUserOptionsFilter.SYSTEM_ADMIN) {
+            options[UserSearchOptions.ROLE] = SearchUserOptionsFilter.SYSTEM_ADMIN;
+        } else if (filter === SearchUserOptionsFilter.ALLOW_INACTIVE) {
+            options[SearchUserOptionsFilter.ALLOW_INACTIVE] = true;
+        }
+        return options;
+    }
```
Comment (lieut-data) [inline components/admin_console/system_users/system_users.jsx]: nit: add new line to space out functions here

**Comment by pradeepmurugesan** [comment]
@lieut-data Fixed the review comments. 

I hope you are talking about `setSystemUsersSearch` function.  I had the same feeling. 
There is another iteration to filter out **User with Access Token Permission ** as part of https://github.com/mattermost/mattermost-server/issues/7494. Will refactor when implementing the same.

**Comment by grundleborg** [comment]
Thanks very much for the PR @pradeepmurugesan

**Pull request merged**